docs/08-upgrading/06-upgrading-from-serverpod_auth.md

```diff
@@ -0,0 +1,253 @@
+The new package also makes used of the recently introduced support for `UUID` primary key on all its entities. Thus in addition to migrating from the legacy package to the new ones, one also has to update all their own entities which previously referenced the `UserInfo`’s `id`.
+
+For an existing Serverpod application which makes use of `serverpod_auth`[^1] to upgrade to the new packages, there exists the packages `serverpod_auth_migration` to facilitate moving over all authentication- and user-related data into the new modules.  
+The package `serverpod_auth_backwards_compatibility` was created to support existing clients with legacy sessions and the migration of social logins and email passwords[^2].
+
+Once the one-time migration is complete, the legacy `serverpod_auth` module and the `serverpod_auth_migration` dependencies can be removed. This will also remove the obsolete tables from the database.
+The backwards-compatibility package needs to be kept until all relevant data has been fully migrated.[^3]
+
+ℹ️ The currently provided migration helpers make the following assumptions:
+
+1. That the size of your database is small enough to be migrated in whatever maintenance window you can allot.
+2. That clients either get updated immediately (e.g. Flutter Web) or that an update can be forced (for installed apps), in order to align with breaking changes on the API when switching from the legacy to the new endpoints.  \
+Clients which update will be able to keep running on their existing session, but clients that do not update won’t work anymore.
+
+As there is no urgency to migrate to the new packages for existing applications, the transition should be carefully planned and tested.
+
+## General timeline
+
+The overall timeline to migrate from `serverpod_auth` to the new modules is given below.
+
+1. Add and configure the new auth modules for the desired providers, e.g. [email](../concepts/authentication/setup_new#email).
+2. Add the migration module `serverpod_auth_migration` and configure the migration the server’s `run` method.
+3. Update the `authenticationHandler` to use the new `serverpod_auth_session` package.
+4. Add the `serverpod_auth_backwards_compatibility` module and connect its helpers in the account login methods.
+5. Disable the `serverpod_auth` endpoint.
+6. Update the client to only use the new endpoints and the `SessionManager` from `serverpod_auth_session_flutter`. Ensure that `serverpod_auth_client` and `serverpod_auth_shared_flutter` are not used anymore.  
+   If deploying to an app store with long lead times, prepare this well in advance, so that new updates / downloads will be able to login and register against the new APIs.
+7. Once the updated app is available for users, deploy the backed and force the client to upgrade.
+8. Remove the `serverpod_auth_migration` and `serverpod_auth` module from the server.
```

Comment:
TODO: Also mention the client side and which packages should be removed there (e.g. Flutter `SessionManager`).